test/e2e/crud.go

```diff
@@ -44,49 +50,128 @@ var _ = Describe("Federated", func() {
+					if len(clusterName) == 0 {
+						clusterName = key
+						break
+					}
 				}
-				typeConfig, err := common.GetTypeConfig(client, typeConfigName, f.FederationSystemNamespace())
+				clusterConfig := testClusters[clusterName].Config
+
+				By("Waiting for the test namespace to be created in the selected cluster")
+				kubeClient := kubeclientset.NewForConfigOrDie(clusterConfig)
+				common.WaitForNamespaceOrDie(tl, kubeClient, clusterName, targetObject.GetNamespace(),
+					framework.PollInterval, framework.TestContext.SingleCallTimeout)
+
+				By("Creating an unlabeled resource in the selected cluster")
+				unlabeledObj, err := common.CreateResource(clusterConfig, typeConfig.GetTarget(), targetObject)
 				if err != nil {
-					tl.Fatalf("Error retrieving federatedtypeconfig %q: %v", typeConfigName, err)
+					tl.Fatalf("Failed to create unlabeled resource in cluster %q: %v", clusterName, err)
 				}
+				clusterClient := genericclient.NewForConfigOrDie(clusterConfig)
+				defer clusterClient.Delete(context.TODO(), unlabeledObj, unlabeledObj.GetNamespace(), unlabeledObj.GetName())
 
-				if framework.TestContext.LimitedScope && !typeConfig.GetNamespaced() {
-					framework.Skipf("Federation of cluster-scoped type %s is not supported by a namespaced control plane.", typeConfigName)
+				By("Intitializing a federated resource with placement excluding all clusters")
+				fedObject, err := federate.FederatedResourceFromTargetResource(typeConfig, unlabeledObj)
+				if err != nil {
+					tl.Fatalf("Error generated federated resource: %v", err)
+				}
+				err = util.SetClusterNames(fedObject, []string{})
+				if err != nil {
+					tl.Fatalf("Error setting cluster names for federated resource: %v", err)
 				}
+				fedObject.SetGenerateName("")
 
-				testObjectsFunc := func(namespace string, clusterNames []string) (*unstructured.Unstructured, []interface{}, error) {
-					targetObject, err := common.NewTestTargetObject(typeConfig, namespace, fixture)
-					if err != nil {
-						return nil, nil, err
-					}
-					if typeConfig.GetTarget().Kind == util.NamespaceKind {
-						// Namespace crud testing needs to have the same name as its namespace.
-						targetObject.SetName(namespace)
-						targetObject.SetNamespace(namespace)
+				By("Creating the federated resource")
+				createdObj, err := common.CreateResource(f.KubeConfig(), typeConfig.GetFederatedType(), fedObject)
+				if err != nil {
+					tl.Fatalf("Error creating federated resource: %v", err)
+				}
+				hostClient := genericclient.NewForConfigOrDie(f.KubeConfig())
+				defer hostClient.Delete(context.TODO(), createdObj, createdObj.GetNamespace(), createdObj.GetName())
+
+				waitDuration := 10 * time.Second // Arbitrary amount of time to wait for deletion
+				By(fmt.Sprintf("Checking that the unlabeled resource remains is not deleted within %v", waitDuration))
+				wait.PollImmediate(framework.PollInterval, waitDuration, func() (bool, error) {
+					obj := unlabeledObj.DeepCopy()
```

Comment:
[Query]For my ignorance; reason of using this in place of an empty object.

Reply:
Expediency - I needed an object with the correct group/version/resource and this is simpler than creating a new empty object.

Reply:
Done